### DISCUSSION TITLE
Error - installation of the Version 2.3.90 on Centos CentOS 7 64-bit.Need help to understand why it is failing and how to fix.

### DISCUSSION THREAD

**Discussion by UserOnion** [Security-Onion-Solutions/securityonion#6367] [Unsupported Versions]
Need help to understand why it is failing and how to fix.

Hardware NUC8i7BEH1, BIOS Firmware - 357.86 ( last one on 11/28/21) 
Attempts to install Onion ISO directly on hardware failed with Kernel Panic, but this is different  topic.

Installed CENTOS 7 / 64 Server Core.

Trying to install SO on top of the Centos as listed in https://docs.securityonion.net/en/2.3/installation.html
Installation failed with following error:
_===========
Executing command with retry support: docker pull ghcr.io/security-onion-solutions/so-elastalert:2.3.90
net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
Results: 2.3.90: Pulling from security-onion-solutions/so-elastalert
a330b6cecb98: Pulling fs layer
f4ea433a9572: Pulling fs layer
8d302f6e98a0: Pulling fs layer
ab162a393884: Pulling fs layer
7ac15c35d0b8: Pulling fs layer
7c64e670752e: Pulling fs layer
422501fecfd0: Pulling fs layer
5003e92fc59f: Pulling fs layer
3aeff7e8bd03: Pulling fs layer
fce7984b9f72: Pulling fs layer
b10a794bb1fa: Pulling fs layer
ab162a393884: Waiting
7ac15c35d0b8: Waiting
7c64e670752e: Waiting
422501fecfd0: Waiting
5003e92fc59f: Waiting
3aeff7e8bd03: Waiting
fce7984b9f72: Waiting
b10a794bb1fa: Waiting
f4ea433a9572: Verifying Checksum
f4ea433a9572: Download complete
ab162a393884: Verifying Checksum
ab162a393884: Download complete
7ac15c35d0b8: Verifying Checksum
7ac15c35d0b8: Download complete
7c64e670752e: Verifying Checksum
7c64e670752e: Download complete
422501fecfd0: Retrying in 5 seconds
a330b6cecb98: Retrying in 5 seconds
...................
>>>>> ...  snip
...................
8d302f6e98a0: Retrying in 1 second
422501fecfd0: Downloading (1)
Command failed with exit code 1; will retry in 60 seconds (1 / 5)...
Executing command with retry support: docker pull ghcr.io/security-onion-solutions/so-elastalert:2.3.90
Error response from daemon: Get https://ghcr.io/v2/: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
Results:  (1)
Command failed with exit code 1; will retry in 60 seconds (2 / 5)...
Executing command with retry support: docker pull ghcr.io/security-onion-solutions/so-elastalert:2.3.90
Error response from daemon: Get https://ghcr.io/v2/: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
Results:  (1)
Command failed with exit code 1; will retry in 60 seconds (3 / 5)...
Executing command with retry support: docker pull ghcr.io/security-onion-solutions/so-elastalert:2.3.90
Error response from daemon: Get https://ghcr.io/v2/: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
Results:  (1)
Command failed with exit code 1; will retry in 60 seconds (4 / 5)...
Executing command with retry support: docker pull ghcr.io/security-onion-solutions/so-elastalert:2.3.90
Error response from daemon: Get https://ghcr.io/v2/: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
Results:  (1)
Command failed with exit code 1; will retry in 60 seconds (5 / 5)...
Command continues to fail; giving up.
Could not pull so-elastalert:2.3.90, please ensure connectivity to ghcr.io
Errors detected during setup; skipping post-setup steps to allow for analysis of failures.
====================

Summary :

====================
Security Onion Version: 2.3.90
Node Type: STANDALONE
Hostname: onion
Management NIC: enp0s20f0u2
Management IP: 192.168.1.233
Proxy: N/A
Bond NIC(s):
  - eno1
Home Network(s):
  - 192.168.1.0/24
Access URL: https://192.168.1.233
Allowed IP or Subnet: 192.168.1.0/24
Web User: ***@*** 
Fleet User: ***@***
Enabled Optional Components:
  - OSQUERY
  - WAZUH
  - THEHIVE
  - PLAYBOOK
  - STRELKA
Metadata Tool: ZEEK
IDS Ruleset: ETOPEN
Patch Schedule:
  Type: auto
OS Package Updates: Open
Zeek Processes: 1
Suricata Processes: 1
Elasticsearch Heap Size: 5381m
Logstash Heap Size: 1000m
Logstash Worker Count: 8
Logstash Batch Size: 125
Logstash Input Threads: 1
Elasticsearch Storage Space: 306GB

**Comment by Acewiza** [comment]
If I had a nickle for every time a slow/congested network caused some random problem....

**Reply by UserOnion** [reply]
it is not latency issue. It is failing every time on the same step.

**Comment by dougburks** [comment]
Check your network for firewalls, proxy servers, or other devices that may be interfering with the download.

**Reply by UserOnion** [reply]
If you installing SO this error appears in the end of the installation, before that al other packages downloaded without problem. 
No Proxy, FW or IPS. DNS pointed to 8.8.8.8

**Reply by dougburks** [reply]
If there is nothing on your local network that is interfering, then there could be something in your upstream network provider that is interfering. Try a different network.

Another possibility is that a partition is reaching 100%. Check your partitioning against the requirements here:
https://docs.securityonion.net/en/2.3/partitioning.html
